// @yrpri/webapp/common/yp-share-dialog.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { html } from 'lit';
import { property, customElement } from 'lit/decorators.js';
import { YpBaseElement } from './yp-base-element.js';
import 'share-menu/share-menu.js';
import 'share-menu/targets/facebook.js';
import 'share-menu/targets/email.js';
import 'share-menu/targets/whatsapp.js';
import 'share-menu/targets/linkedin.js';
let YpShareDialog = class YpShareDialog extends YpBaseElement {
    render() {
        return html `
      <share-menu
        class="shareIcon"
        id="shareButton"
        @close="${this.haveSharedContentCallback}"
        .title="${this.label || ''}"
        .url="${this.url || ''}"
        .image="${this.image || ''}"
      >
        <share-target-email></share-target-email>
        <share-target-facebook></share-target-facebook>
        <share-target-twitter></share-target-twitter>
        <share-target-whatsapp></share-target-whatsapp>
        <share-target-linkedin></share-target-linkedin>
      </share-menu>
    `;
    }
    async open(url, label, image, haveSharedContentCallback) {
        this.url = url;
        this.label = label;
        this.haveSharedContentCallback = haveSharedContentCallback;
        this.image = image;
        this.requestUpdate();
        await this.updateComplete;
        this.$$('#shareButton') /*ShareMenu*/.share();
    }
};
__decorate([
    property({ type: Object })
], YpShareDialog.prototype, "haveSharedContentCallback", void 0);
__decorate([
    property({ type: String })
], YpShareDialog.prototype, "url", void 0);
__decorate([
    property({ type: String })
], YpShareDialog.prototype, "image", void 0);
__decorate([
    property({ type: String })
], YpShareDialog.prototype, "label", void 0);
YpShareDialog = __decorate([
    customElement('yp-share-dialog')
], YpShareDialog);
export { YpShareDialog };
